pkg/transparentproxy/kubernetes/kubernetes.go

```diff
@@ -22,26 +22,36 @@ import (
 
 	kube_core "k8s.io/api/core/v1"
 
+	runtime_k8s "github.com/kumahq/kuma/pkg/config/plugins/runtime/k8s"
 	"github.com/kumahq/kuma/pkg/plugins/runtime/k8s/metadata"
 	"github.com/kumahq/kuma/pkg/transparentproxy/config"
 )
 
 type PodRedirect struct {
-	BuiltinDNSEnabled                  bool
-	BuiltinDNSPort                     uint32
-	ExcludeOutboundPorts               string
-	RedirectPortOutbound               uint32
-	RedirectInbound                    bool
-	ExcludeInboundPorts                string
-	RedirectPortInbound                uint32
-	RedirectPortInboundV6              uint32
-	UID                                string
-	ExperimentalTransparentProxyEngine bool
+	BuiltinDNSEnabled                        bool
+	BuiltinDNSPort                           uint32
+	ExcludeOutboundPorts                     string
+	RedirectPortOutbound                     uint32
+	RedirectInbound                          bool
+	ExcludeInboundPorts                      string
+	RedirectPortInbound                      uint32
+	RedirectPortInboundV6                    uint32
+	UID                                      string
+	ExperimentalTransparentProxyEngine       bool
+	TransparentProxyEnableEbpf               bool
+	TransparentProxyEbpfBPFFSPath            string
+	TransparentProxyEbpfInstanceIPEnvVarName string
+	TransparentProxyEbpfProgramsSourcePath   string
 }
 
-func NewPodRedirectForPod(pod *kube_core.Pod) (*PodRedirect, error) {
+func NewPodRedirectForPod(pod *kube_core.Pod, cfg runtime_k8s.Injector) (*PodRedirect, error) {
```

Comment:
This is a strange dependency. Isn't these configurations usually passed in from cp to init-containers by using pod annotations? I believe in this can the annotation is always present so there's no need to read from the cfg default

Reply:
addressed